### PR TITLE
FIX: disables send btn while uploads in progress

### PR DESCRIPTION
**Pull request by jjaffeux**
Before this fix you could press send while upload was in progress and lose it as it was not yet uploaded.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/mixins/uppy-upload.js [hunk 1]
```diff
@@ -318,6 +318,7 @@ export default Mixin.create(UppyS3Multipart, ExtendableUploader, {
   },
 
   _triggerInProgressUploadsEvent() {
+    this.onProgressUploadsChanged?.(this.inProgressUploads);
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/mixins/uppy-upload.js:321]: I prefer to not rely on an appEvent if I can avoid this.

File: plugins/chat/assets/javascripts/discourse/components/chat-composer.js [hunk 2]
```diff
@@ -116,7 +117,9 @@ export default class ChatComposer extends Component {
   }
 
   get sendEnabled() {
-    return this.hasContent && !this.pane.sending;
+    return (
+      this.hasContent && !this.pane.sending && !this.inProgressUploadsCount > 0
```
Comment (CvX) [inline plugins/chat/assets/javascripts/discourse/components/chat-composer.js:121]:
```suggestion
      this.hasContent && !this.pane.sending && this.inProgressUploadsCount === 0
```

**Review by CvX — APPROVED** [review]

**Comment by CvX** [comment]
Hm, test failures 👀

**Comment by jjaffeux** [comment]
> Hm, test failures 👀

wow, never seen one before

**Pull request merged**